sonic-xcvrd/xcvrd/dom/utilities/db/utils.py

```diff
@@ -1,59 +1,198 @@
+from datetime import datetime
 from swsscommon import swsscommon
+from xcvrd.xcvrd_utilities.utils import XCVRDUtils
 
 class DBUtils:
     """
     This class contains utility functions to interact with the redis database.
     """
-    def __init__(self, logger):
+    NEVER = "never"
+    NOT_AVAILABLE = "N/A"
+
+    def __init__(self, sfp_obj_dict, port_mapping, task_stopping_event, logger):
+        self.sfp_obj_dict = sfp_obj_dict
+        self.port_mapping = port_mapping
+        self.task_stopping_event = task_stopping_event
+        self.xcvrd_utils = XCVRDUtils(sfp_obj_dict, logger)
         self.logger = logger
 
-    """
-    Updates the metadata tables for flag table
-    As part of the metadata update, the following tables are updated:
-    - Change Count Table
-    - Last Set Time Table
-    - Last Clear Time Table
-    """
-    def update_flag_metadata_tables(self, logical_port_name, field_name, current_value,
+    def post_diagnostic_values_to_db(self, logical_port_name, table, get_values_func,
+                                     db_cache=None, beautify_func=None, enable_flat_memory_check=False):
+        """
+        Posts the diagnostic values to the database.
+
+        Args:
+            logical_port_name (str): Logical port name.
+            table (object): Database table object.
+            get_values_func (function): Function to get diagnostic values.
+            db_cache (dict, optional): Cache for diagnostic values.
+            beautify_func (function, optional): Function to beautify the diagnostic values. Defaults to self.beautify_info_dict.
+            enable_flat_memory_check (bool, optional): Flag to check for flat memory support. Defaults to False.
+        """
+        if self.task_stopping_event.is_set():
+            return
+
+        pport_list = self.port_mapping.get_logical_to_physical(logical_port_name)
+        if not pport_list:
+            self.logger.log_error(f"Post port diagnostic values to db failed for {logical_port_name} "
+                                   "as no physical port found")
+            return
+        physical_port = pport_list[0]
+
+        if physical_port not in self.sfp_obj_dict:
+            self.logger.log_error(f"Post port diagnostic values to db failed for {logical_port_name} "
+                                   "as no sfp object found")
+            return
+
+        if not self.xcvrd_utils.get_transceiver_presence(physical_port):
+            return
+
+        if enable_flat_memory_check and self.xcvrd_utils.is_transceiver_flat_memory(physical_port):
+            return
```

Comment:
I see many functions share the same logic of this, is it possible to encapsulate this code block with a function?